src/net_processing.cpp

```diff
@@ -2675,7 +2675,7 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         // Inbound peers send us their version message when they connect.
         // We send our version message in response.
         if (pfrom.IsInboundConn()) {
-            PushNodeVersion(pfrom, *peer);
+            PushNodeVersion(pfrom, peer->m_tx_relay == nullptr);
```

Comment:
Duplicating `peer->m_tx_relay == nullptr` at both call sites rather than just leaving it in the function seems kind of backwards? Also seems demonstrably a bit bug-prone, presumably warranting a `/*block_relay_only=*/` at least.

Reply:
Alternatively, pass the tx_relay pointer?

Reply:
Reverted change so this passes in a `const Peer&`